eras/conway/impl/src/Cardano/Ledger/Conway/TxBody.hs

```diff
@@ -451,16 +456,45 @@ totalDRepDeposits pp txb = nDReps <×> depositPerDRep
     nDReps = getSum @Int (foldMap' (\case RegDRepTxCert {} -> 1; _ -> 0) (txb ^. certsTxBodyL))
     depositPerDRep = pp ^. ppDRepDepositL
 
+-- | Compute all the deposits in a TxBody
+--   This includes deposits for
+--   1) registering Stake
+--   2) registering a StakePool
+--   3) registering a DRep
+--   4) making a Proposal
+-- This is the contribution of a TxBody towards the total 'Obligations' of the system
+-- See the Tyeps and Functions Cardano.Ledger.CertState(Obligations,obligationCertState) for more information.
 totalTxDepositsConway ::
   Crypto c =>
   PParams (ConwayEra c) ->
   CertState (ConwayEra c) ->
   TxBody (ConwayEra c) ->
   Coin
 totalTxDepositsConway pp cs txb =
-  totalTxDepositsShelley pp cs txb
-    <> totalProposalDeposits pp txb
-    <> totalDRepDeposits pp txb
+  totalTxDepositsShelley pp cs txb -- Stake and StakePool
+    <> totalProposalDeposits pp txb -- only in Conway and later
+    <> totalDRepDeposits pp txb -- only in Conway and later
+
+drepRefundsConway :: ConwayEraTxBody era => CertState era -> TxBody era -> Coin
+drepRefundsConway certState txb = Coin (snd (foldl' accum (vsDReps (certVState certState), 0) (txb ^. certsTxBodyL)))
+  where
+    accum pair@(currMap, currRefund) (UnRegDRepTxCert cred (Coin n)) = case (Map.lookup cred currMap) of
+      (Just (DRepState _ _ (Coin m))) | n == m -> (Map.delete cred currMap, currRefund + m)
+      _ -> pair
+    accum pair@(currMap, currRefund) (RegDRepTxCert cred c _) = case (Map.lookup cred currMap) of
+      Nothing -> (Map.insert cred (DRepState (EpochNo 0) SNothing c) currMap, currRefund)
+      -- we are only ever going to look at 'c', so the bogus epoch (EpochNo 0) and anchor (SNothing) are OK.
```

Comment:
This is horrible. If you ever start using bogus values, try to think if it is possible to rewrite a function that doesn't rely on bogus values. In this case it is totally possible. 

In any case, there is a better implementation of this logic right here:

https://github.com/input-output-hk/cardano-ledger/blob/47f7b053553a505e7ab3b6cbf4550c1a199215f7/eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs#L113-L122

Reply:
Redone as suggested, but without PParams.